views-core/src/main/java/io/micronaut/views/http/ResponseBodySwapper.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.views.http;
+
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.core.order.Ordered;
+import io.micronaut.http.HttpRequest;
+import io.micronaut.http.HttpResponse;
+
+import java.util.Optional;
+
+/**
+ * A functional interface to swap the response's body with a new one.
+ * @param <B> The response body
+ * @author Sergio del Amo
+ * @since 6.0.0
+ */
+@FunctionalInterface
+public interface ResponseBodySwapper<B> extends Ordered {
```

Comment:
does this need to be public?

Reply:
Yes, it's used here and in the turbo module (in a different package)